icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -173,18 +173,27 @@ UnicodeSet* initAlpha(UErrorCode& status) {
 UnicodeSet* initDigits(UErrorCode& status) {
     UnicodeSet* result = new UnicodeSet(UnicodeString("[:number:]"), status);
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
     result->freeze();
     return result;
 }
 
 UnicodeSet* initNameStartChars(UErrorCode& status) {
     if (U_FAILURE(status)) {
-        return {};
+        return nullptr;
     }
 
-    UnicodeSet* result = new UnicodeSet(*unisets::getImpl(unisets::ALPHA));
+    unisets::gUnicodeSets[unisets::ALPHA] = initAlpha(status);
```

Comment:
why not just 
```
UnicodeSet* isAlpha = unisets::gUnicodeSets[unisets::ALPHA] =  initAlpha(status);
if (U_FAILURE(status)) {
        return nullptr;
}
```

Reply:
Fixed in [e2990d2](https://github.com/unicode-org/icu/pull/3306/commits/e2990d2b9ccf2f31832348fa29d0f36ff4c3002f)